internal/broker/client.go

```diff
@@ -245,10 +292,18 @@ func (c *Client) formatDeprovisionUrl(instance internal.Instance) (string, error
 	return fmt.Sprintf(deprovisionTmpl, c.brokerConfig.URL, instancesURL, instance.InstanceID, kymaClassID, instance.ServicePlanID), nil
 }
 
-func (c *Client) executeRequestWithPoll(method, url string, expectedStatus int, body io.Reader, responseBody interface{}) error {
-	request, err := http.NewRequest(method, url, body)
+func (c *Client) formatUnbindUrl(binding internal.Binding) (string, error) {
+	if len(binding.InstanceID) == 0 {
+		return "", fmt.Errorf("empty InstanceID")
+	}
+
+	return fmt.Sprintf(unbindTmpl, c.brokerConfig.URL, instancesURL, binding.InstanceID, binding.ID), nil
+}
+
+func (c *Client) executeRequest(method, url string, expectedStatus int, requestBody io.Reader, responseBody interface{}) error {
```

Comment:
As far as I remember you pass http.StatusOK in the invocation.
So http.StatusGone would we treated as not acceptable?

Reply:
correct, `http.StatusGone` would be unexpected. I think it's ok to return an error in such case and leave it up to the method caller how to handle the error. If the error included HTTP status code, then the caller could act accordingly.

Reply:
Discussed in direct communication.